File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/options/KeyMapping.java
package pize.tests.minecraft.game.options;

import pize.io.glfw.Key;

public enum KeyMapping{

    FORWARD   (Key.W),
    LEFT      (Key.A),
    BACK      (Key.S),
    RIGHT     (Key.F),
    JUMP      (Key.SPACE),
    SNEAK     (Key.LEFT_SHIFT),
    SPRINT    (Key.LEFT_CONTROL),

    CHAT      (Key.T),
    ZOOM      (Key.C),

    FULLSCREEN(Key.F11),
    SCREENSHOT(Key.F2);


    private final Key defaultKey;

    KeyMapping(Key defaultKey){
        this.defaultKey = defaultKey;
    }

    public Key getDefaultKey(){
        return defaultKey;
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/options/KeyMapping.kt
package pize.tests.minecraft.game.options

import pize.io.glfw.Key

enum class KeyMapping private constructor(val defaultKey: Key) {
    FORWARD(Key.W),
    LEFT(Key.A),
    BACK(Key.S),
    RIGHT(Key.F),
    JUMP(Key.SPACE),
    SNEAK(Key.LEFT_SHIFT),
    SPRINT(Key.LEFT_CONTROL),
    CHAT(Key.T),
    ZOOM(Key.C),
    FULLSCREEN(Key.F11),
    SCREENSHOT(Key.F2)

}
